### PR TITLE
Setup and add integration tests

### PR DESCRIPTION
**Pull request by gcatanese**
**Description**
Setup integration tests:

- define env variabled (api keys, etc..) to inject
- define base integration testing class
- create few tests (Checkout, LEM, BCL)
- add doc

Tests do not run on GitHub, they will be enabled in a different PR.

**Tested scenarios**
Run all tests locally

### CODE REVIEW

File: src/integration-test/java/com/adyen/BaseIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+/** Shared client and local configuration access for integration tests. */
+public abstract class BaseIntegrationTest {
+
+  private static final String CONFIGURATION_RESOURCE = "/config.properties";
+  private static final String API_KEY_PROPERTY = "API_LIBRARIES_ADYEN_API_KEY";
+  private static final String LEM_API_KEY_PROPERTY = "API_LIBRARIES_ADYEN_LEM_API_KEY";
+  private static final String BCL_API_KEY_PROPERTY = "API_LIBRARIES_ADYEN_BCL_API_KEY";
+  private static final Properties PROPERTIES = loadProperties();
+  private final Map<String, Client> clients = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (gcatanese) [reply src/integration-test/java/com/adyen/BaseIntegrationTest.java:26]: JUnit 5 uses per-method test instance lifecycle by default, so the `clients` map is never shared across threads — each test method gets its own `BaseIntegrationTest` instance. Even with parallel execution enabled, there is no concurrent access to the same map instance. `HashMap` with `computeIfAbsent` is correct here.

File: src/test/java/com/adyen/BaseCloudDeviceIntegrationTest.java [hunk 2]
```diff
@@ -52,7 +52,9 @@ private Properties getProperties() {
     if (properties == null) {
       properties = new Properties();
       try (InputStream inputStream =
-          BaseIntegrationTest.class.getClassLoader().getResourceAsStream("config.properties")) {
+          BaseCloudDeviceIntegrationTest.class
+              .getClassLoader()
+              .getResourceAsStream("config.properties")) {
```
[the remark this answers is not in the document]
Comment (gcatanese) [reply src/test/java/com/adyen/BaseCloudDeviceIntegrationTest.java:57]: Valid point, though the bootstrap classloader edge case is practically impossible for application test code. That said, this is pre-existing code — the PR only renamed the class. I'll leave it as-is for this PR to keep the diff focused on the integration-test setup.
